### PR TITLE
Merge Dev to main branch for muilple changes 

### LINKED ISSUE #70: [feature]: Saving the embeddings to the vector db
The embeddings can be stored into vector db.

## Note

We do not need to consider update the embeddings currently.

### PR DESCRIPTION
**Pull request by Micost**
**Description**

This PR fixes #70 #68 #67 #42 #41 
**Notes for Reviewers**

**[Signed commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: backend/Dockerfile_alternative [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM gclub/skywardai:v0.1.2
```
Comment (Aisuko) [inline backend/Dockerfile_alternative:1]: Is this one necessary?

File: backend/Dockerfile [hunk 2]
```diff
@@ -1,13 +1,15 @@
-FROM gclub/skywardai:v0.1.2
+FROM gclub/skywardai:v0.1.3
 
 WORKDIR /app
 
 EXPOSE 8000
 
+# Install Python dependencies
+COPY ./requirements.txt ./
```
Comment (Aisuko) [inline backend/Dockerfile:8]: We install all the requirements in base image. There may not need to install the dependencies again

File: backend/src/api/routes/chat.py [hunk 3]
```diff
@@ -35,6 +35,8 @@ async def chat(
         session_id = chat_in_msg.sessionId
     await chat_repo.create_chat_history(session_id=session_id, is_bot_msg=False, message=chat_in_msg.message)
     response_msg = await rag_chat_repo.get_response(session_id=session_id, input_msg=chat_in_msg.message)
+    print("----------------")
```
Comment (Aisuko) [inline backend/src/api/routes/chat.py:38]:
```suggestion
```

File: backend/src/api/routes/chat.py [hunk 4]
```diff
@@ -35,6 +35,8 @@ async def chat(
...
+    print(response_msg)
```
Comment (Aisuko) [inline backend/src/api/routes/chat.py:39]:
```suggestion
```

File: backend/src/config/settings/const.py [hunk 5]
```diff
@@ -1,5 +1,23 @@
 # Uploaded file path
 UPLOAD_FILE_PATH = "./uploaded_files/"
 MAX_SQL_LENGTH = 200
-DEFAULT_MODEL = "all-MiniLM-L6-v2"
+
+DEFAULT_COLLECTION = "default_collection"
+DEFAULT_DIM = 384
 CHAT_COMTEXT = "Melbourne is the capital city of the Australian state of Victoria.It is known for its diverse and vibrant cultural scene.The city is famous for its coffee culture, with numerous cafes scattered throughout.Melbourne is home to iconic landmarks like the Royal Exhibition Building and Flinders Street Station.The Yarra River runs through the heart of the city, providing a picturesque setting.The Melbourne Cricket Ground (MCG) is a historic sports venue and a key part of the city's identity.The city hosts various events, including the Australian Open, Melbourne Fashion Week, and the Melbourne International Comedy Festival.Melbourne's street art scene is renowned, with vibrant murals adorning many laneways.The Queen Victoria Market is a popular spot for fresh produce, local crafts, and diverse international cuisines.Melbourne is often considered one of the most livable cities globally, offering a high quality of life."
+# DEFAULT MODELS
```
Comment (Aisuko) [inline backend/src/config/settings/const.py:8]: How do we deal with these contents?

Comment (Micost) [reply backend/src/config/settings/const.py:8]: It is the initial context  works when no csv file uploaded

File: backend/Dockerfile_alternative (hunk 1 above)
Comment (Micost) [reply backend/Dockerfile_alternative]: removed

File: backend/src/api/routes/chat.py (hunk 4 above)
Comment (Micost) [reply backend/src/api/routes/chat.py]: removed as the other debug flags

**Comment by Aisuko** [comment]
LGTM

**Pull request merged**